oci/private/pull.bzl

```diff
@@ -479,78 +487,57 @@ oci_alias = repository_rule(
     },
 )
 
-_latest_build = """\
-load("@aspect_bazel_lib//lib:jq.bzl", "jq")
-load("@bazel_skylib//rules:write_file.bzl", "write_file")
-
-jq(
-    name = "platforms",
-    srcs = ["manifest_list.json"],
-    filter = "[(.manifests // [])[] | .platform | .os + \\"/\\" + .architecture]",
-    # Print without newlines because it's too hard to indent that to fit under the generated
-    # starlark code below.
-    args = ["--compact-output"],
-)
-
-jq(
-    name = "mediaType",
-    srcs = ["manifest_list.json"],
-    filter = ".mediaType",
-    args = ["--raw-output"],
-)
-
+_LATEST_BUILD = """\
 sh_binary(
     name = "pin",
     srcs = ["pin.sh"],
-    data = [
-        ":mediaType",
-        ":platforms",
-        "@bazel_tools//tools/bash/runfiles",
-    ],
 )
 """
 
-_pin_sh = """\
+_PIN_SH = """\
 #!/usr/bin/env bash
-{rlocation}
 
-mediaType="$(cat $(rlocation {name}/mediaType.json))"
 echo -e "Replace your '{reponame}' declaration with the following:\n"
 
 cat <<EOF
 oci_pull(
     name = "{reponame}",
     digest = "sha256:{digest}",
     image = "{registry}/{repository}",
+    {optional_platforms}
+)
 EOF
-
-[[ $mediaType == "{manifestListType}" ]] && cat <<EOF
```

Comment:
this was broken as it assumed that the manifest list type is always `application/vnd.oci.image.index.v1+json` which is not true for docker v2 images.  

also, it's overkill to make these calculations at runtime. it's easier to calculate platforms in the repository rule itself. see below.